### ISSUE TITLE
[camera] App throws exception when opening gallery 

### ISSUE THREAD

**Issue by vimaxwell** [flutter/flutter#89352]
## Steps to Reproduce
This is errors on my firebase crashlytics in camera: ^0.9.2+1.

**Expected results:**
It doesn't happen in old version of camera ^.8.x

**Actual results:**
```
Fatal Exception: java.lang.IllegalArgumentException: No handler given, and current thread has no looper!
       at android.hardware.camera2.impl.CameraDeviceImpl.checkHandler(CameraDeviceImpl.java:2272)
       at android.hardware.camera2.impl.CameraDeviceImpl.checkHandler(CameraDeviceImpl.java:2287)
       at android.hardware.camera2.impl.CameraCaptureSessionImpl.setRepeatingRequest(CameraCaptureSessionImpl.java:306)
       at io.flutter.plugins.camera.Camera.refreshPreviewCaptureSession(Camera.java:26)
       at io.flutter.plugins.camera.Camera.access$600(Camera.java)
       at io.flutter.plugins.camera.Camera$2.onConfigured(Camera.java:50)
       at android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy.lambda$onConfigured$0$CallbackProxies$SessionStateCallbackProxy(CallbackProxies.java:53)
       at android.hardware.camera2.impl.-$$Lambda$CallbackProxies$SessionStateCallbackProxy$soW0qC12Osypoky6AfL3P2-TeDw.run(-.java:4)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:923)
```

```
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.2 20G95 darwin-x64, locale en)
    • Flutter version 2.2.3 at /Users/maxwell/development/flutter
    • Framework revision f4abaa0735 (9 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/maxwell/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_301-b09)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.59.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (2 available)
    • Aris (mobile) • BDB00022215 • android-arm64  • Android 11 (API 30)
    • Chrome (web)  • chrome      • web-javascript • Google Chrome 92.0.4515.159
    ! Error: iPhone is not connected. Xcode will continue when iPhone is connected. (code -13)
```

</details>

**Comment by TahaTesser** [comment]
Hi @vimaxwell 
Can you please logs run your app in release mode? Try to take picture pics or perform camera actions

Also please provide your list of affected devices?
Thank you

**Comment by vimaxwell** [comment]
@TahaTesser 
It happens on many devices. I can't reproduce it on my device. It just showed up on my firebase and google play publisher.
![image](https://user-images.githubusercontent.com/19898639/131991718-c1ea4b5b-8131-414f-9844-ada5b6105085.png)

**Comment by vimaxwell** [comment]
@TahaTesser
However, if I test on debug mode on my demo app https://github.com/vimaxwell/camera_test, it shows similar error. I think it is same root cause.
### Steps to produce
after previewing camera, I click on the gallery image icon (on the right botom) to navigate to gallery, then errors appears on console.

```
I/CameraManagerGlobal( 5538): Connecting to camera service
W/CameraManagerGlobal( 5538): [soar.cts] ignore the status update of camera: 2
I/Camera  ( 5538): startPreview
I/Camera  ( 5538): Updating builder settings
D/Camera  ( 5538): Updating builder with feature: ExposureLockFeature
D/Camera  ( 5538): Updating builder with feature: ExposurePointFeature
D/Camera  ( 5538): Updating builder with feature: ZoomLevelFeature
D/Camera  ( 5538): Updating builder with feature: AutoFocusFeature
D/Camera  ( 5538): Updating builder with feature: NoiseReductionFeature
I/Camera  ( 5538): updateNoiseReduction | currentSetting: fast
D/Camera  ( 5538): Updating builder with feature: FocusPointFeature
D/Camera  ( 5538): Updating builder with feature: ResolutionFeature
D/Camera  ( 5538): Updating builder with feature: SensorOrientationFeature
D/Camera  ( 5538): Updating builder with feature: FlashFeature
D/Camera  ( 5538): Updating builder with feature: ExposureOffsetFeature
D/Camera  ( 5538): Updating builder with feature: FpsRangeFeature
I/Gralloc2( 5538): Adding additional valid usage bits: 0x200000
I/Camera  ( 5538): dispose
I/Camera  ( 5538): close
I/Camera  ( 5538): closeCaptureSession
W/MessageQueue( 5538): Handler (android.os.Handler) {f1c01bf} sending message to a Handler on a dead thread
W/MessageQueue( 5538): java.lang.IllegalStateException: Handler (android.os.Handler) {f1c01bf} sending message to a Handler on a dead thread
W/MessageQueue( 5538): 	at android.os.MessageQueue.enqueueMessage(MessageQueue.java:546)
W/MessageQueue( 5538): 	at android.os.Handler.enqueueMessage(Handler.java:745)
W/MessageQueue( 5538): 	at android.os.Handler.sendMessageAtTime(Handler.java:697)
W/MessageQueue( 5538): 	at android.os.Handler.sendMessageDelayed(Handler.java:667)
W/MessageQueue( 5538): 	at android.os.Handler.post(Handler.java:395)
W/MessageQueue( 5538): 	at android.hardware.camera2.impl.CameraDeviceImpl$CameraHandlerExecutor.execute(CameraDeviceImpl.java:2330)
W/MessageQueue( 5538): 	at android.hardware.camera2.impl.CameraDeviceImpl.close(CameraDeviceImpl.java:1176)
W/MessageQueue( 5538): 	at io.flutter.plugins.camera.Camera.close(Camera.java:1096)
W/MessageQueue( 5538): 	at io.flutter.plugins.camera.Camera.dispose(Camera.java:1119)
W/MessageQueue( 5538): 	at io.flutter.plugins.camera.MethodCallHandlerImpl.onMethodCall(MethodCallHandlerImpl.java:361)
W/MessageQueue( 5538): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
W/MessageQueue( 5538): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
W/MessageQueue( 5538): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:818)
W/MessageQueue( 5538): 	at android.os.MessageQueue.nativePollOnce(Native Method)
W/MessageQueue( 5538): 	at android.os.MessageQueue.next(MessageQueue.java:326)
```

**Comment by vimaxwell** [comment]
@BeMacized 
I hope you can help to confirm and fix this one.

**Comment by TahaTesser** [comment]
Hi @vimaxwell 
Thanks for the details, I can reproduce it based on  https://github.com/flutter/flutter/issues/89352#issuecomment-912445398

<details> 
<summary>logs</summary> 
	
```console
[+4902 ms] I/Camera  ( 3510): runPictureAutoFocus
[   +1 ms] I/Camera  ( 3510): lockAutoFocus
[   +2 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
[  +23 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
[   +4 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
[  +16 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
[  +17 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
[  +26 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
[  +34 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
[  +24 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 4 | aeState: 2
[   +1 ms] I/Camera  ( 3510): captureStillPicture
[   +1 ms] D/Camera  ( 3510): Updating builder with feature: ExposureLockFeature
[   +1 ms] D/Camera  ( 3510): Updating builder with feature: ExposurePointFeature
[   +1 ms] D/Camera  ( 3510): Updating builder with feature: ZoomLevelFeature
[        ] D/Camera  ( 3510): Updating builder with feature: AutoFocusFeature
[        ] D/Camera  ( 3510): Updating builder with feature: NoiseReductionFeature
[        ] I/Camera  ( 3510): updateNoiseReduction | currentSetting: fast
[        ] D/Camera  ( 3510): Updating builder with feature: FocusPointFeature
[        ] D/Camera  ( 3510): Updating builder with feature: ResolutionFeature
[        ] D/Camera  ( 3510): Updating builder with feature: SensorOrientationFeature
[        ] D/Camera  ( 3510): Updating builder with feature: FlashFeature
[        ] D/Camera  ( 3510): Updating builder with feature: ExposureOffsetFeature
[        ] D/Camera  ( 3510): Updating builder with feature: FpsRangeFeature
[  +38 ms] I/Camera  ( 3510): sending capture request
[  +11 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
[  +83 ms] I/Camera  ( 3510): unlockAutoFocus
[  +13 ms] D/CameraCaptureCallback( 3510): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
[  +46 ms] I/Camera  ( 3510): onImageAvailable
[   +2 ms] D/qdgralloc( 3510): GetYUVPlaneInfo: Invalid format passed: 0x21
[+12226 ms] W/Choreographer( 3510): Frame time is 0.141802 ms in the future!  Check that graphics HAL is generating
vsync timestamps using the correct timebase.
[+15523 ms] I/Timeline( 3510): Timeline: Activity_launch_request time:176434090
[  +47 ms] W/MessageQueue( 3510): Handler (android.os.Handler) {f584be8} sending message to a Handler on a dead thread
[   +1 ms] W/MessageQueue( 3510): java.lang.IllegalStateException: Handler (android.os.Handler) {f584be8} sending
message to a Handler on a dead thread
[   +1 ms] W/MessageQueue( 3510):       at android.os.MessageQueue.enqueueMessage(MessageQueue.java:563)
[        ] W/MessageQueue( 3510):       at android.os.Handler.enqueueMessage(Handler.java:778)
[        ] W/MessageQueue( 3510):       at android.os.Handler.sendMessageAtTime(Handler.java:727)
[        ] W/MessageQueue( 3510):       at android.os.Handler.sendMessageDelayed(Handler.java:697)
[        ] W/MessageQueue( 3510):       at android.os.Handler.post(Handler.java:427)
[        ] W/MessageQueue( 3510):       at
android.hardware.camera2.impl.CameraDeviceImpl$CameraHandlerExecutor.execute(CameraDeviceImpl.java:2252)
[        ] W/MessageQueue( 3510):       at
android.hardware.camera2.impl.CameraDeviceImpl$CameraDeviceCallbacks.onResultReceived(CameraDeviceImpl.java:2160)
[        ] W/MessageQueue( 3510):       at
android.hardware.camera2.ICameraDeviceCallbacks$Stub.onTransact(ICameraDeviceCallbacks.java:180)
[        ] W/MessageQueue( 3510):       at android.os.Binder.execTransactInternal(Binder.java:1162)
[        ] W/MessageQueue( 3510):       at android.os.Binder.execTransact(Binder.java:1126)
[        ] W/MessageQueue( 3510): Handler (android.os.Handler) {f584be8} sending message to a Handler on a dead thread
[        ] W/MessageQueue( 3510): java.lang.IllegalStateException: Handler (android.os.Handler) {f584be8} sending
message to a Handler on a dead thread
[        ] W/MessageQueue( 3510):       at android.os.MessageQueue.enqueueMessage(MessageQueue.java:563)
[        ] W/MessageQueue( 3510):       at android.os.Handler.enqueueMessage(Handler.java:778)
[        ] W/MessageQueue( 3510):       at android.os.Handler.sendMessageAtTime(Handler.java:727)
[        ] W/MessageQueue( 3510):       at android.os.Handler.sendMessageDelayed(Handler.java:697)
[        ] W/MessageQueue( 3510):       at android.os.Handler.post(Handler.java:427)
[        ] W/MessageQueue( 3510):       at
android.hardware.camera2.impl.CameraDeviceImpl$CameraHandlerExecutor.execute(CameraDeviceImpl.java:2252)
[        ] W/MessageQueue( 3510):       at
android.hardware.camera2.impl.CameraDeviceImpl$CameraDeviceCallbacks.onCaptureStarted(CameraDeviceImpl.java:1964)
[        ] W/MessageQueue( 3510):       at
android.hardware.camera2.ICameraDeviceCallbacks$Stub.onTransact(ICameraDeviceCallbacks.java:158)
[        ] W/MessageQueue( 3510):       at android.os.Binder.execTransactInternal(Binder.java:1162)
[        ] W/MessageQueue( 3510):       at android.os.Binder.execTransact(Binder.java:1126)
[        ] I/Camera  ( 3510): dispose
[        ] I/Camera  ( 3510): close
[        ] I/Camera  ( 3510): closeCaptureSession
[ +254 ms] W/BpBinder( 3510): PerfMonitor binderTransact :  time=256ms
interface=android.hardware.camera2.ICameraDeviceUser code=1
[   +3 ms] W/MessageQueue( 3510): Handler (android.os.Handler) {f584be8} sending message to a Handler on a dead thread
[   +2 ms] W/MessageQueue( 3510): java.lang.IllegalStateException: Handler (android.os.Handler) {f584be8} sending
message to a Handler on a dead thread
[   +2 ms] W/MessageQueue( 3510):       at android.os.MessageQueue.enqueueMessage(MessageQueue.java:563)
[   +1 ms] W/MessageQueue( 3510):       at android.os.Handler.enqueueMessage(Handler.java:778)
[   +1 ms] W/MessageQueue( 3510):       at android.os.Handler.sendMessageAtTime(Handler.java:727)
[   +1 ms] W/MessageQueue( 3510):       at android.os.Handler.sendMessageDelayed(Handler.java:697)
[        ] W/MessageQueue( 3510):       at android.os.Handler.post(Handler.java:427)
[        ] W/MessageQueue( 3510):       at
android.hardware.camera2.impl.CameraDeviceImpl$CameraHandlerExecutor.execute(CameraDeviceImpl.java:2252)
[        ] W/MessageQueue( 3510):       at android.hardware.camera2.impl.CameraDeviceImpl.close(CameraDeviceImpl.java:1363)
[        ] W/MessageQueue( 3510):       at io.flutter.plugins.camera.Camera.close(Camera.java:1096)
[        ] W/MessageQueue( 3510):       at io.flutter.plugins.camera.Camera.dispose(Camera.java:1119)
[        ] W/MessageQueue( 3510):       at
io.flutter.plugins.camera.MethodCallHandlerImpl.onMethodCall(MethodCallHandlerImpl.java:361)
[        ] W/MessageQueue( 3510):       at
io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
[        ] W/MessageQueue( 3510):       at
io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
[        ] W/MessageQueue( 3510):       at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:818)
[        ] W/MessageQueue( 3510):       at android.os.MessageQueue.nativePollOnce(Native Method)
[        ] W/MessageQueue( 3510):       at android.os.MessageQueue.next(MessageQueue.java:335)
[        ] W/MessageQueue( 3510):       at android.os.Looper.loop(Looper.java:193)
[        ] W/MessageQueue( 3510):       at android.app.ActivityThread.main(ActivityThread.java:8061)
[        ] W/MessageQueue( 3510):       at java.lang.reflect.Method.invoke(Native Method)
[        ] W/MessageQueue( 3510):       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
[        ] W/MessageQueue( 3510):       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
[+3231 ms] I/Camera  ( 3510): close
[  +31 ms] E/libc    ( 3510): Access denied finding property "persist.vendor.camera.privapp.list"
[   +1 ms] W/mple.camera_app( 3510): type=1400 audit(0.0:498929): avc: denied { read } for
name="u:object_r:vendor_persist_camera_prop:s0" dev="tmpfs" ino=28868 scontext=u:r:untrusted_app:s0:c62,c257,c512,c768
tcontext=u:object_r:vendor_persist_camera_prop:s0 tclass=file permissive=0 app=com.example.camera_app
[  +13 ms] I/Camera  ( 3510): startPreview
[  +91 ms] I/Camera  ( 3510): Updating builder settings
[   +2 ms] D/Camera  ( 3510): Updating builder with feature: ExposureLockFeature
[   +2 ms] D/Camera  ( 3510): Updating builder with feature: ExposurePointFeature
[   +2 ms] D/Camera  ( 3510): Updating builder with feature: ZoomLevelFeature
[   +1 ms] D/Camera  ( 3510): Updating builder with feature: AutoFocusFeature
[        ] D/Camera  ( 3510): Updating builder with feature: NoiseReductionFeature
[        ] I/Camera  ( 3510): updateNoiseReduction | currentSetting: fast
[        ] D/Camera  ( 3510): Updating builder with feature: FocusPointFeature
[        ] D/Camera  ( 3510): Updating builder with feature: ResolutionFeature
[        ] D/Camera  ( 3510): Updating builder with feature: SensorOrientationFeature
[        ] D/Camera  ( 3510): Updating builder with feature: FlashFeature
[        ] D/Camera  ( 3510): Updating builder with feature: ExposureOffsetFeature
[        ] D/Camera  ( 3510): Updating builder with feature: FpsRangeFeature
[+26306 ms] W/Choreographer( 3510): Frame time is 0.171079 ms in the future!  Check that graphics HAL is generating
vsync timestamps using the correct timebase.
```
	
</details>


|  `stable`  | `master` |
| :---: |:---: |
|  ❗️  |   ❗️  | 

Check `flutter doctor -v` outputs for each channel below

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.3, on Microsoft Windows [Version 10.0.22000.176], locale en-US)
    • Flutter version 2.2.3 at C:\Users\Taha\Code\flutter_stable
    • Framework revision f4abaa0735 (9 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\Taha\Code\android-sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_SDK_ROOT = C:\Users\Taha\Code\android-sdk
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[✓] VS Code (version 1.60.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.26.0

[✓] Connected device (4 available)
    • Redmi K20 Pro (mobile) • def0ad20 • android-arm64  • Android 11 (API 30)
    • Windows (desktop)      • windows  • windows-x64    • Microsoft Windows [Version 10.0.22000.176]
    • Chrome (web)           • chrome   • web-javascript • Google Chrome 93.0.4577.63
    • Edge (web)             • edge     • web-javascript • Microsoft Edge 92.0.902.84

! Doctor found issues in 1 category.
```

```console
[✓] Flutter (Channel master, 2.6.0-1.0.pre.170, on Microsoft Windows [Version 10.0.22000.176], locale en-US)
    • Flutter version 2.6.0-1.0.pre.170 at C:\Users\Taha\Code\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4069c6355b (7 hours ago), 2021-09-03 02:11:05 -0400
    • Engine revision d6f6a0fe90
    • Dart version 2.15.0 (build 2.15.0-69.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\Taha\Code\android-sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_SDK_ROOT = C:\Users\Taha\Code\android-sdk
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31624.102
    • Windows 10 SDK version 10.0.19041.0

[✓] Android Studio (version 2020.3)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.0)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.26.0

[✓] Connected device (5 available)
    • Redmi K20 Pro (mobile)  • def0ad20 • android-arm64   • Android 11 (API 30)
    • Windows (desktop)       • windows  • windows-x64     • Microsoft Windows [Version 10.0.22000.176]
    • Windows (UWP) (desktop) • winuwp   • windows-uwp-x64 •
    • Chrome (web)            • chrome   • web-javascript  • Google Chrome 93.0.4577.63
    • Edge (web)              • edge     • web-javascript  • Microsoft Edge 92.0.902.84

• No issues found!
```
		
</details>

✅: No Issue ❗️: Issue reproduced

**Comment by BeMacized** [comment]
> @BeMacized
> I hope you can help to confirm and fix this one.

@vimaxwell I was not able to reproduce the exception throw in your initial post, however I did for the one you commented in [#89352 (comment)](https://github.com/flutter/flutter/issues/89352#issuecomment-912445398).  I implemented a potential fix in flutter/plugins#4319. Would you be able to check if this fixes your issue?

**Comment by vimaxwell** [comment]
> > @BeMacized
> > I hope you can help to confirm and fix this one.
> 
> @vimaxwell I was not able to reproduce the exception in your initial post, however I did for the one you commented in [#89352 (comment)](https://github.com/flutter/flutter/issues/89352#issuecomment-912445398). I implemented a potential fix in [flutter/plugins#4319](https://github.com/flutter/plugins/pull/4319). Would you be able to check if this fixes your issue?

thanks. this issue is fixed

**Comment by AlexV525** [comment]
I'm raising the priority to P2, since the issue will cause the whole app lifecycle hook fail when using a camera instance.

**Comment by carman247** [comment]
Think some of our users are also starting to get this error after I've upgraded to camera 0.9.2+2

```
Fatal Exception: java.lang.IllegalArgumentException
No handler given, and current thread has no looper!
android.hardware.camera2.impl.CameraDeviceImpl.checkHandler (CameraDeviceImpl.java:2410)
android.hardware.camera2.impl.CameraDeviceImpl.checkHandler (CameraDeviceImpl.java:2425)
android.hardware.camera2.impl.CameraCaptureSessionImpl.setRepeatingRequest (CameraCaptureSessionImpl.java:290)
io.flutter.plugins.camera.Camera.refreshPreviewCaptureSession (Camera.java:26)
io.flutter.plugins.camera.Camera.access$600 (Camera.java)
io.flutter.plugins.camera.Camera$2.onConfigured (Camera.java:50)
android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy.lambda$onConfigured$0 (CallbackProxies.java:53)
android.hardware.camera2.impl.-$$Lambda$CallbackProxies$SessionStateCallbackProxy$soW0qC12Osypoky6AfL3P2-TeDw.run (-.java:4)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
````

### Affected devices so far
```
Galaxy Note10+ 5G
Galaxy S20 FE 5G
Galaxy A10
```

**Comment by Hixie** [comment]
The original reporter says this is solved (https://github.com/flutter/flutter/issues/89352#issuecomment-914905221). Should this be closed and new bugs filed for the issues people are running into?

**Comment by AlexV525** [comment]
> The original reporter says this is solved ([#89352 (comment)](https://github.com/flutter/flutter/issues/89352#issuecomment-914905221)). Should this be closed and new bugs filed for the issues people are running into?

Maybe he means this issue will be resolved by the PR?

**Comment by Hixie** [comment]
Ah, got it.

@BeMacized any news on your PR?
